lib/brain.js

```diff
@@ -2567,23 +2568,25 @@ Brain.prototype._physicalDispense = function _physicalDispense () {
       this._transitionState('fiatComplete', {tx: displayTx})
       // this.waitForCollection()
     })
-    .then(() => Promise.delay(60000))
+    .then(() => Bluebird.delay(60000))
     .then(() => {
       const doComplete = this.state === 'fiatComplete' && this.tx.id === txId
 
       if (doComplete) {
         emit('billDispenserCollected')
         return this._completed()
-      } else {
-        throw new Error(`Unknown state ${this.state} !== 'fiatComplete' or ${this.tx.id} !== ${txId}`)
       }
+
+      throw new Error(`Unknown state ${this.state} !== 'fiatComplete' or ${this.tx.id} !== ${txId}`)
     })
     .catch(err => {
       emit('billDispenserCollected')
 
       this.fastUpdateTx({
-        error: err.message,
-        errorCode: err.statusCode || null
+        error: _.isEmpty(err.name)
+          ? err.message
+          : _.join([err.name, err.message], ':'),
```

Comment:
Have you tested this code? lodash fp has reversed arguments, usually the array is at the end.

https://github.com/lodash/lodash/wiki/FP-Guide

Reply:
ok thanks